app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt

```diff
@@ -86,11 +87,13 @@ class SearchFragment : Fragment(), UserInteractionHandler {
     ): View? {
         val activity = activity as HomeActivity
         val args by navArgs<SearchFragmentArgs>()
-        val session = args.sessionId
-            ?.let(requireComponents.core.sessionManager::findSessionById)
+
+        val tabId = args.sessionId
+        val tab = tabId?.let { requireComponents.core.store.state.findTab(it) }
 
         val view = inflater.inflate(R.layout.fragment_search, container, false)
-        val url = session?.url.orEmpty()
+        val url = tab?.content?.url ?: ""
```

Comment:
nit: 
```suggestion
        val url = tab?.content?.url.orEmpty()
```

As we only use searchTerms to create the state, we could just move it into the `SearchFragmentState` constructor.

```kt
SearchFragmentState(
  searchTerms = tab?.content?.searchTerms.orEmpty(),
  ...
)
```